src/metabase/public_settings.clj

```diff
@@ -70,12 +70,60 @@
   :audit   :getter
   :default true)
 
+(defsetting site-uuid
+  ;; Don't i18n this docstring because it's not user-facing! :)
+  "Unique identifier used for this instance of {0}. This is set once and only once the first time it is fetched via
+  its magic getter. Nice!"
+  :visibility :authenticated
+  :base       setting/uuid-nonce-base
+  :doc        false)
+
+(defsetting upgrade-threshold
+  (deferred-tru "Threshold (value in 0-100) indicating at which treshold it should offer an upgrade to the latest major version.")
+  :visibility :internal
+  :export?    false
+  :type       :integer
+  :setter     :none
+  :getter     (fn []
+                (-> (site-uuid) hash (mod 100))))
```

Comment:
I mentioned that in Slack, but maybe we don't want this number to be THAT stable and adding a version info (or something like this) to re-hash our distribution after every upgrade is the better way?

Reply:
updated! good idea